pipelines/main/misc/fullsrcdist.yml

```diff
@@ -0,0 +1,33 @@
+steps:
+  - group: "Full Source Dist" # TODO: delete this line
+  # - group: "Check" # TODO: uncomment this line
+    steps:
+      - label: "fullsrcdist"
+        key: "fullsrcdist"
+        soft_fail: true # TODO: delete this line
+        plugins:
+          - JuliaCI/external-buildkite#v1:
+              version: "./.buildkite-external-version"
+              repo_url: "https://github.com/JuliaCI/julia-buildkite"
+          - JuliaCI/julia#v1:
+              # Drop default "registries" directory, so it is not persisted from execution to execution
+              persist_depot_dirs: packages,artifacts,compiled
+              version: '1.6'
+          - staticfloat/sandbox#v1:
+              rootfs_url: https://github.com/JuliaCI/rootfs-images/releases/download/v4.8/package_linux.x86_64.tar.gz
+              rootfs_treehash: "2a058481b567f0e91b9aa3ce4ad4f09e6419355a"
+              workspaces:
+                - "/cache/repos:/cache/repos"
+          - JuliaCI/julia#v1:
+              # Drop default "registries" directory, so it is not persisted from execution to execution
+              persist_depot_dirs: packages,artifacts,compiled
+              version: '1.6'
+        timeout_in_minutes: 10
```

Comment:
I think `full-source-dist` may take more than 10 minutes... I've never timed it but anecdotally it feels like an eternity. It does a source build of all dependencies then Julia then all documentation.

Reply:
Wait, it actually builds Julia? I thought it just downloads the source code of the dependencies.

Reply:
Yeah haha. It builds Julia so that it can use the built binary to build the documentation.

Reply:
😬

Reply:
Hmmm. A full from-source build of the dependencies and Julia takes more than 30 minutes on our fastest CI machines. See e.g. https://buildkite.com/julialang/julia-master-scheduled/builds/159#01847a53-374d-441c-90cc-c4dcfef6ae22

Maybe we should move the `full-source-dist` job to the once-daily scheduled pipeline. That's where the existing `USE_BINARYBUILDER=0` job lives.

Reply:
Anyway, let's move this to the once-daily scheduled pipeline (https://buildkite.com/julialang/julia-master-scheduled).

That pipeline runs once a day, so if there's a regression on Julia master, we catch it pretty quickly. But we don't run it on PRs, so we save some CI time that way.

Reply:
We should use the same trick that we use elsewhere for the documentation; download a prebuilt Julia, unpack it in the correct place.  Then this won't rebuild Julia because it will already exist.

Reply:
I'm now downloading a pre-built Julia, but it seems to insist on building everything (including LLVM) from source.

Reply:
@ararslan What's the difference between `make full-source-dist` and `make full-source-dist USE_BINARYBUILDER=0`? Shouldn't they both produce the same output? Presumably we shouldn't have to build LLVM from source just to be able to build the docs and download the source tarballs of all dependencies, right?

Reply:
`USE_BINARYBUILDER=0` ensures that the source code of the dependencies is included in the resulting tarball alongside Julia's source code. It's meant to be platform-agnostic and buildable without network access, which IIUC isn't possible with the default JLL-based dependencies.

Reply:
Sure, but shouldn't `make full-source-dist` do that regardless of the value of `USE_BINARYBUILDER`?

I guess what I'm getting at is, do we need to actually do a full from-source build of LLVM etc just to build this tarball? AFAICT, the tarball needs the following contents:
1. Source code of Julia
2. Source code of all of Julia's dependencies
3. HTML documentation

It seems to me that we can use a regular Julia build to build the HTML docs, and then download the source code of all the dependencies, and then stick everything together.

Reply:
We definitely could do that, the Makefile just isn't set up that way